test/e2e_node/init_containers_test.go

```diff
@@ -36,69 +36,101 @@ import (
 	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
 )
 
-type containerType byte
 
-const (
-	containerTypeInvalid containerType = iota
-	containerTypeInit
-	containerTypeRegular
-)
-
-type containerTestConfig struct {
-	Name     string
-	Type     containerType
+type execCommand struct {
 	ExitCode int
 	Delay    int
 }
 
-func (c containerTestConfig) Command() []string {
+func ExecCommand(name string, c execCommand) []string {
 	var cmd bytes.Buffer
 	// all outputs are in the format of:
 	// time-since-boot timestamp container-name message
 
 	// The busybox time command doesn't support sub-second display. uptime displays in hundredths of a second, so we
 	// include both and use time since boot for relative ordering
 	timeCmd := "`date +%s` `cat /proc/uptime | awk '{print $1}'`"
-	fmt.Fprintf(&cmd, "echo %s '%s Starting' >> /dev/termination-log; ", timeCmd, c.Name)
-	fmt.Fprintf(&cmd, "echo %s '%s Delaying %d' >> /dev/termination-log; ", timeCmd, c.Name, c.Delay)
+	containerLog := fmt.Sprintf("/persistent/%s.log", name)
+
+	fmt.Fprintf(&cmd, "touch %s; ", containerLog)
+	fmt.Fprintf(&cmd, "cat %s >> /dev/termination-log; ", containerLog)
+
+	fmt.Fprintf(&cmd, "echo %s '%s Starting' | tee -a %s >> /dev/termination-log; ", timeCmd, name, containerLog)
```

Comment:
Nice, I like the use of tee instead of just catting the persistent log at the end.